projects/storefrontlib/src/cms-components/cart/add-to-cart/add-to-cart.component.ts

```diff
@@ -75,13 +111,29 @@ export class AddToCartComponent implements OnInit, OnDestroy {
   protected setStockInfo(product: Product): void {
     this.quantity = 1;
     this.hasStock = Boolean(
-      product.stock && product.stock.stockLevelStatus !== 'outOfStock'
+      product.stock && product.stock?.stockLevelStatus !== 'outOfStock'
     );
+
+    this.inventoryThreshold = product.stock?.isValueRounded ?? false;
+
     if (this.hasStock && product.stock?.stockLevel) {
       this.maxQuantity = product.stock.stockLevel;
     }
   }
 
+  getInventory(): string {
+    // When backoffice forces 'In Stock' status, DO NOT display stock level info.
+    if (this.hasStock) {
+      // Don't show stock level if product forced to be in stock.
+      const quantityDisplay = this.maxQuantity
+        ? this.maxQuantity.toString()
+        : '';
+      return this.inventoryThreshold ? quantityDisplay + '+' : quantityDisplay;
+    } else {
+      return '';
+    }
+  }
```

Comment:
Now I see this logic spread in this method and in the template.

Idea:
- we create an observable called `stockInfo$`
- it contains the complete text to display in template (translated in stock or out of stock or the particular stock value)

What do you think? Do you see any problems with implementing that?